runtime/auth/aws-signing-common/build.gradle.kts

```diff
@@ -17,6 +19,14 @@ kotlin {
             }
         }
 
+        commonTest {
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-test:$coroutinesVersion")
+                implementation(project(":runtime:auth:aws-signing-tests"))
+                implementation(project(":runtime:auth:aws-signing-default"))
```

Comment:
I had to add the `aws-signing-default` dependency to use [`DefaultAwsSigner`](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/auth/aws-signing-default/common/src/aws/smithy/kotlin/runtime/auth/awssigning/DefaultAwsSigner.kt#L13) in my test cases. Similarly the `aws-signing-tests` dependency was so I could use [`testSigningConfig`](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/auth/aws-signing-tests/common/src/aws/smithy/kotlin/runtime/auth/awssigning/tests/SigningTestUtil.kt#L11) from there. I can remove the aws-signing-tests dependency since it's a pretty simple construction, but I figured I should use the existing util variable rather than duplicate it. But I'm not sure how to remove the dependency on aws-signing-default while also continuing to use the `DefaultAwsSigner`.

Reply:
What if we just move the tests for `AwsChunkedByteReadChannel` into `aws-signing-tests`? This would remove the pseudo circular dependency (`aws-signing-default` depends on `aws-signing-common` for the commonMain sourceSet) and allow us to cover both default and crt based signers. 

I realize it's slightly odd to not have the test live with the source but I think in this case it may be the best option.

Thoughts?

Reply:
It is a good idea, but complicates things because `AwsChunkedByteReadChannel` is `internal` to `aws-signing-common` and can't be accessed from `aws-signing-tests`. I can change the visibility, but I think internal is the best choice here.

I was looking into Kotlin's behavior around `internal` and saw it is a group of files compiled together, is there any way to group these two modules together during compilation to allow access in aws-signing-tests?

Reply:
As far as we are concerned `internal` and `public` with `@InternalApi` are nearly the same from a support/maintenance policy (obviously `internal` is better since it is not accessible at all whereas the latter can be accessed with compiler flags). 

I think I'd just mark it as `public` and annotate it with `@InternalApi` but happy to see what others think.